examples/search_examples/public/components/app.tsx

```diff
@@ -0,0 +1,303 @@
+                <br />
+                Is it Cool? {String((response as IMyStrategyResponse).cool)}
+              </EuiText>
+            );
+            notifications.toasts.addSuccess({
+              title: 'Query result',
+              text: mountReactNode(message),
+            });
+            searchSubscription$.unsubscribe();
+          } else if (response.isPartial && !response.isRunning) {
+            // TODO: Make response error status clearer
+            notifications.toasts.addWarning('An error has occurred');
+            searchSubscription$.unsubscribe();
+          }
+        },
+        error: () => {
+          notifications.toasts.addDanger('Failed to run search');
+        },
+      });
+  };
+
+  const onClickHandler = () => {
+    doAsyncSearch();
+  };
+
+  const onMyStrategyClickHandler = () => {
+    doAsyncSearch('myStrategy');
+  };
+
+  if (!indexPattern) return null;
+
+  return (
+    <Router basename={basename}>
+      <I18nProvider>
+        <>
+          <navigation.ui.TopNavMenu
+            appName={PLUGIN_ID}
+            showSearchBar={true}
+            useDefaultBehaviors={true}
+            indexPatterns={indexPattern ? [indexPattern] : undefined}
+          />
+          <EuiPage restrictWidth="1000px">
+            <EuiPageBody>
+              <EuiPageHeader>
+                <EuiTitle size="l">
+                  <h1>
+                    <FormattedMessage
+                      id="searchExamples.helloWorldText"
+                      defaultMessage="{name}"
+                      values={{ name: PLUGIN_NAME }}
+                    />
+                  </h1>
+                </EuiTitle>
+              </EuiPageHeader>
+              <EuiPageContent>
+                <EuiPageContentBody>
+                  <EuiText>
+                    <EuiFlexGrid columns={1}>
+                      <EuiFlexItem>
+                        <EuiFormLabel>Index Pattern</EuiFormLabel>
+                        <IndexPatternSelect
+                          savedObjectsClient={savedObjectsClient}
+                          placeholder={i18n.translate(
+                            'backgroundSessionExample.selectIndexPatternPlaceholder',
+                            {
+                              defaultMessage: 'Select index pattern',
+                            }
+                          )}
+                          indexPatternId={indexPattern?.id || ''}
+                          onChange={async (newIndexPatternId: any) => {
+                            const newIndexPattern = await data.indexPatterns.get(newIndexPatternId);
+                            setIndexPattern(newIndexPattern);
+                          }}
+                          isClearable={false}
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiFormLabel>Numeric Fields</EuiFormLabel>
+                        <EuiComboBox
```

Comment:
@lukasolson I think we should export a `IndexPatternFieldSelect` from `data` to make it easier to use